cmd/algorelay/relayCmd.go

```diff
@@ -415,50 +432,85 @@ func ensureRelayStatus(checkOnly bool, relay eb.Relay, nameDomain string, srvDom
 		}
 
 		if len(matches) == 0 {
-			return fmt.Errorf("no %s SRV entries found mapping to %s in '%s'", use, target, srvDomain)
+			return 0, fmt.Errorf("no %s SRV entries found mapping to %s in '%s'", use, target, srvDomain)
 		}
 
 		if len(matches) > 1 {
-			return fmt.Errorf("multiple %s SRV entries found in the chain mapping to %s", use, target)
+			return len(matches), fmt.Errorf("multiple %s SRV entries found in the chain mapping to %s", use, target)
 		}
 
 		if matches[0].name != topmost || matches[0].port != port {
-			return fmt.Errorf("existing %s SRV record mapped to intermediate DNS name or wrong port (wanted %s:%d, found %s:%d)",
+			return len(matches), fmt.Errorf("existing %s SRV record mapped to intermediate DNS name or wrong port (wanted %s:%d, found %s:%d)",
 				use, topmost, port, matches[0].name, matches[0].port)
 		}
-		return nil
+		return len(matches), nil
 	}
 
-	err = ensureEntry("bootstrap", ctx.bootstrap.entries, port)
-	if err != nil {
-		if checkOnly {
-			return
-		}
-
-		// Add SRV entry to map to our DNSAlias
-		err = addSRVRecord(ctx.bootstrap.networkName, topmost, port, ctx.bootstrap.shortName, ctx.srvZoneID)
-		if err != nil {
-			return
-		}
-		fmt.Printf("[%d] Added boostrap SRV Record: %s:%d\n", relay.ID, targetDomainAlias, port)
-	}
-
-	err = ensureEntry("metrics", ctx.metrics.entries, metricsPort)
-	if relay.MetricsEnabled {
+	var matchCount int
+	matchCount, err = ensureEntry("algobootstrap", ctx.bootstrap.entries, port)
+	if relay.CheckSuccess {
 		if err != nil {
 			if checkOnly {
 				return
 			}
 
-			// Add SRV entry for metrics
-			err = addSRVRecord(ctx.metrics.networkName, topmost, metricsPort, ctx.metrics.shortName, ctx.srvZoneID)
+			// Add SRV entry to map to our DNSAlias
+			err = addSRVRecord(ctx.bootstrap.networkName, topmost, port, ctx.bootstrap.shortName, ctx.srvZoneID)
 			if err != nil {
 				return
 			}
-			fmt.Printf("[%d] Added metrics SRV Record: %s:%d\n", relay.ID, targetDomainAlias, metricsPort)
+			fmt.Printf("[%d] Added boostrap SRV Record: %s:%d\n", relay.ID, targetDomainAlias, port)
+		}
+	} else {
+		if matchCount > 0 {
+			err = clearSRVRecord(ctx.bootstrap.networkName, topmost, ctx.bootstrap.shortName, ctx.srvZoneID)
```

Comment:
Similarly, checkOnly is not considered for clearSRVRecord (in all 3 cases below). 
Is this what was intended?